src/stream/mod.rs

```diff
@@ -880,7 +882,7 @@ pub trait Stream {
     /// This method is only available when the `use_std` feature of this
     /// library is activated, and it is activated by default.
     #[cfg(feature = "use_std")]
-    fn buffered(self, amt: usize) -> Buffered<Self>
+    fn buffered(self, amt: usize) -> Buffered<FuturesOrdered<<<Self as Stream>::Item as IntoFuture>::Future>, Self>
```

Comment:
In any case I'd prefer to keep all the separate types for future flexibility. The internals can look like this (sharing impls) but the API should continue to expose unique types.